src/common/options.rs

```diff
@@ -2224,6 +2224,52 @@ impl AsRef<str> for Model {
     }
 }
 
+impl From<String> for Model {
+    fn from(value: String) -> Self {
+        match &*value {
+            "nova-2" => Self::Nova2,
+            "nova" => Self::Nova,
+            "enhanced" => Self::Enhanced,
+            "base" => Self::Base,
```

Comment:
Let's explicitly handle `*-general` model specifiers. 
```suggestion
            "nova-2" | "nova-2-general" => Self::Nova2,
            "nova" | "nova-general"  => Self::Nova,
            "enhanced" | "enhanced-general" => Self::Enhanced,
            "base" | "base-general" => Self::Base,
```